teal/packer/display.tl

```diff
@@ -265,75 +184,81 @@ local function diff(disp: Display)
       return
     end
     vim.schedule(function()
-      open_preview(commit, lines)
+      if not lines or #lines < 1 then
+        log.warn 'No diff available'
+        return
+      end
+      local buf = open_win(true)
+      api.nvim_buf_set_lines(buf, 0, -1, false, lines)
+      api.nvim_buf_set_name(buf, commit)
+      vim.keymap.set('n', 'q', '<cmd>close!<cr>', { buffer = buf, silent = true, nowait = true })
+      vim.bo[buf].filetype = 'git'
     end)
   end)
 end
 
-local function make_update_msg(symbol: string, status: string, plugin_name: string, plugin: Plugin): string
-  return fmt(' %s %s %s: %s..%s', symbol, status, plugin_name, plugin.revs[1], plugin.revs[2])
-end
-
 --- Update the text of the display buffer
-local function set_lines(disp: Display, start_idx: integer, end_idx: integer, lines: {string})
-  if not valid_display(disp) then
-    return
-  end
+local function set_lines(disp: Display, srow: integer, erow: integer, lines: {string})
   vim.bo[disp.buf].modifiable = true
-  local ok, err = pcall(api.nvim_buf_set_lines, disp.buf, start_idx, end_idx, true, lines) as boolean, string
-  if not ok then
-    print(vim.inspect(err))
-    error(string.format('Could not set lines %d-%d: %s\n%s', start_idx, end_idx, vim.inspect(err), debug.traceback()))
-  end
+  api.nvim_buf_set_lines(disp.buf, srow, erow, true, lines)
   vim.bo[disp.buf].modifiable = false
 end
 
-local function toggle_update(disp: Display)
-  if not disp:is_previewing() then
-    return
-  end
-  local plugin_name = find_nearest_plugin(disp)
-  local item = disp.items[plugin_name]
-  if not item then
-    log.warn 'Plugin not available!'
+local function get_task_region(self: Display, plugin: string): integer, integer
+  local mark = self.items[plugin].mark
+
+  if not mark then
     return
   end
 
-  item.ignore_update = not item.ignore_update
+  local info = api.nvim_buf_get_extmark_by_id(self.buf, ns, mark, {details=true})
 
-  local mark_ids = disp.marks[plugin_name]
-  local start_idx = api.nvim_buf_get_extmark_by_id(disp.buf, disp.ns, mark_ids.start, {})[1]
-  local symbol: string
-  local status_msg: string
-  if item.ignore_update then
-    status_msg = [[Won't update]]
-    symbol = config.display.item_sym
-  else
-    status_msg = 'Can update'
-    symbol = config.display.done_sym
+  local srow = info[1]
+  local erow = info[3].end_row or srow
+
+  -- TODO(lewis6991): sometimes the end_row will be lower and start_row. Could
```

Comment:
I think (not sure?) this is why we had the `pcall` for setting lines earlier.

Reply:
Gotta be an extmark bug I think. An `end_row` should never be above the `start_row`.